thunder/core/interpreter.py

```diff
@@ -2671,6 +2671,25 @@ def create_namedtuple(typename: str, field_names: str, **kwargs):
     return namedtuple_type
 
 
+def _type_call_lookaside(wrapped_typ, *args, **kwargs):
+    typ = unwrap(wrapped_typ)
+    if not hasattr(typ, "__new__"):
+        raise NotImplementedError(
+            f"Don't know how to interpret a callable with type {type(typ)} without a __new__ method"
+        )
+    obj = _interpret_call(typ.__new__, wrapped_typ, *args, **kwargs)
+    if obj is INTERPRETER_SIGNALS.EXCEPTION_RAISED:
+        return obj
+
+    wrapped_init = _interpret_call(getattr, obj, wrap_const("__init__"))
+    assert not isinstance(wrapped_init, INTERPRETER_SIGNALS)
+    populate_attribute_wrapper(wrapped_init, "__self__", obj)
+    res = _interpret_call(wrapped_init, *args, **kwargs)
+    if res is INTERPRETER_SIGNALS.EXCEPTION_RAISED:
+        return res
+    return obj
```

Comment:
I am a bit confused with these lines. After successful init we still return the result of new?

Reply:
Yes, init returns None.

Reply:
Yeah, makes sense :)